### PR TITLE
Add TextInput controlled selection prop on Android

### PR DESCRIPTION
**Pull request by janicduplessis**
Android PR for TextInput selection, based on the iOS implementation in #8958.

** Test plan **
Tested using the text selection example in UIExplorer.

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @JoelMarcey and @javache to be potential reviewers.

**Comment by javache** [comment]
cc @foghina

**Comment by foghina** [comment]
@andreicoman11

**Comment by facebook-github-bot-6** [comment]
@janicduplessis updated the pull request.

**Comment by janicduplessis** [comment]
iOS support just landed, rebased on top of it so this is ready for review.

ping @andreicoman11

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -675,6 +674,11 @@ const TextInput = React.createClass({
       children = <Text>{children}</Text>;
     }
 
+    let selection = this.props.selection;
```
Comment (andreicoman11) [inline Libraries/Components/TextInput/TextInput.js]: Any reason why this is different from the way iOS handles selection? I'd prefer it to be the same.
Also, here `end` is `selection.end` which is null, while on iOS `end` is `selection.start`.

File: Examples/UIExplorer/js/TextInputExample.android.js [hunk 2]
```diff
@@ -623,4 +710,22 @@ exports.examples = [
...
+    title: 'Text selection & cursor placement',
+    render: function() {
+      return (
+        <View>
+          <SelectionExample
+            style={styles.default}
+            value="text selection can be changed"
+          />
+          <SelectionExample
+            multiline
```
Comment (andreicoman11) [inline Examples/UIExplorer/js/TextInputExample.android.js]: what does this do?

**Comment by andreicoman11** [comment]
Lgtm, just a few inlines

**Comment by facebook-github-bot-7** [comment]
@janicduplessis updated the pull request - [view changes](https://github.com/facebook/react-native/pull/8962/files/5819aab11a9be59b4aa71f4daa2d380e1780172b..366275b2149f033406f25eb1981eb984bc71702c)

File: Libraries/Components/TextInput/TextInput.js (hunk 1 above)
Comment (janicduplessis) [inline Libraries/Components/TextInput/TextInput.js]: Good catch that should be like iOS, fixed

File: Examples/UIExplorer/js/TextInputExample.android.js (hunk 2 above)
Comment (janicduplessis) [inline Examples/UIExplorer/js/TextInputExample.android.js]: Not sure exectly what you mean here. The example is a copy paste of the iOS one (for some reason this example is probably the only one split in 2 files).

The examples is just a bunch of buttons to change the selection for a single line and multiline text input.

Comment (andreicoman11) [inline Examples/UIExplorer/js/TextInputExample.android.js]: Sorry, I wasn't very clear, and maybe my JS knowledge isn't up to par here.
I'm confused as to what "multiline" does here. Is it a prop? Is it a style? Is it necessary?

Comment (janicduplessis) [inline Examples/UIExplorer/js/TextInputExample.android.js]: Oh it's just a prop that is used to tell if we want to render that example with a normal or multiline text input. So what it does is renders the same example but one with single line and the other with multiline input.

**Comment by andreicoman11** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**